### ISSUE TITLE
pkgconfig for plugins are missing dependencies

### ISSUE THREAD

**Issue by glebm** [openscenegraph/OpenSceneGraph#1052]
E.g. pkgconfig for the "png" plugin should specify a dependency on png, similar for jpeg and freetype2.

This matters when plugins are linked statically.

**Comment by robertosfield** [comment]
You'll need to be be more specific about what you mean by pkgconfig and how you are using it and what how it relates to your usage of the OSG, as right now I'm a bit stumped by exactly what you are talking about.

**Comment by glebm** [comment]
Here is an example of a correct pkg-config file that is installed by OSG:

```
$ cat /usr/lib/x86_64-linux-gnu/pkgconfig/openscenegraph-osgShadow.pc 
# pkg-config source file

prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib/x86_64-linux-gnu
includedir=${prefix}/include

Name: openscenegraph-osgShadow
Description: Shadow techniques library for Openscenegraph
Version: 3.6.5
Requires: openscenegraph-osgUtil openscenegraph-osgDB openscenegraph-osg openthreads
Conflicts:
Libs: -L${libdir} -losgShadow
Cflags: -I${includedir}
```

This file correctly specifies the dependencies of `osgShadow` in the requires section.

The installation of this file is handled here:

https://github.com/openscenegraph/OpenSceneGraph/blob/34a1d8bc9bba5c415c4ff590b3ea5229fa876ba8/CMakeModules/ModuleInstall.cmake#L113-L114

However, for OSG plugins, pkg-config files are not installed at all.

---

Some background: I want to improve CMake's built-in OSG functions to use pkg-config if present.

This should make static linking easier on platforms that support pkg-config.

Currently, for statically linking OSG, one must be careful with the order of OSG libraries and plugins. For example: https://gitlab.com/OpenMW/openmw/-/merge_requests/640/diffs

**Comment by glebm** [comment]
This is more or less low priority for OpenMW, now that we've figured out how to link OSG in all scenarios (dynamic/static x system/vendored)

**Comment by openscenegraph** [comment]
On Sun, 7 Mar 2021 at 14:49, Gleb Mazovetskiy <notifications@github.com>
wrote:

> This is more or less low priority for OpenMW, now that we've figured out
> how to link OSG in all scenarios (dynamic/static x system/vendored)
>

The pkg-config support was a submission from quite a few years back, it's
not something I personally contributed to.

Going forward using cmake config support would be my preference as support
across platforms.  There is support merged with master, a contribution from
the OSG community.

The VSG has CMake config support that I implemented so I have more
knowledge about cmake config support so should have a better chance of
maintaining it.  This would not be something for the 3.6 branch though as
it's in maintenance more.

**Comment by glebm** [comment]
> There is support merged with master

Ah, fantastic, CMake configs are even better in OpenMW's case!
Closing this then
